main.go

```diff
@@ -148,7 +148,7 @@ func sigHandler(w http.ResponseWriter, r *http.Request) {
 	output, err := callAutograph(auth, input, xff)
 	if err != nil {
 		log.WithFields(log.Fields{"rid": rid, "input_sha256": inputSha256}).Error(err)
-		http.Error(w, "failed to call autograph for signature", http.StatusBadGateway)
+		httpError(w, r, http.StatusBadGateway, "failed to call autograph for signature")
```

Comment:
this tries to read the input again but I think that's OK

Reply:
yeah 

>  A successful Copy returns err == nil, not err == EOF. Because Copy is defined to read from src until EOF, it does not treat an EOF from Read as an error to be reported. 

https://golang.org/pkg/io/#Copy